modules/swagger-jaxrs/src/test/java/io/swagger/GenericsTest.java

```diff
@@ -228,4 +216,12 @@ public void scanModelWithGenericType() {
         assertNotNull(properties.get("id"));
         assertNotNull(properties.get("name"));
     }
+
+    @Test(description = "scan model implementing interface with Generic Type")
+    public void scanModelImplInterfaceWithGenericType() {
+        final Swagger swagger = new Reader(new Swagger()).read(UserApiRouteImpl.class);
+        assertNotNull(swagger);
+        assertEquals(((BodyParameter) swagger.getPath("/api/users2").getPost().getParameters().get(0))
+                .getSchema().getReference(), "#/definitions/UserEntity");
```

Comment:
This is the test that would fail if the change is not present. The failing value would be `#/definitions/AbstractEntity`